### PR TITLE
feat: add support for u64,i64,f64 fields in term aggregation

### PR DESCRIPTION
**Pull request by PSeitz**

### CODE REVIEW

File: src/aggregation/mod.rs [hunk 1]
```diff
@@ -271,6 +271,22 @@ pub enum Key {
     /// `f64` key
     F64(f64),
 }
+impl Eq for Key {}
+impl std::hash::Hash for Key {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        core::mem::discriminant(self).hash(state);
```
Comment (PSeitz) [inline src/aggregation/mod.rs:277]: This won't work well if someone indexes different values of `f64::NAN` and then does an aggregation on it. 
Do we care about this use case?

**Comment by PSeitz** [comment]
https://github.com/quickwit-oss/tantivy/issues/1689

File: src/aggregation/mod.rs (hunk 1 above)
Comment (fulmicoton) [reply src/aggregation/mod.rs:277]: Let's forbid NaN entirely. If you really want to keep them we can work at least map them upon input to whatever `u64_to_f64(0u64)` is

Comment (fulmicoton) [reply src/aggregation/mod.rs:277]: TIL about Discriminant

Comment (fulmicoton) [inline src/aggregation/mod.rs:277]: You forgot to hash the value isn't it?

File: src/aggregation/mod.rs [hunk 2]
```diff
@@ -271,6 +271,22 @@ pub enum Key {
...
+    }
+}
+
+impl PartialEq for Key {
```
Comment (fulmicoton) [inline src/aggregation/mod.rs:285]: Could this be dervied?

File: src/aggregation/mod.rs (hunk 1 above)
Comment (PSeitz) [reply src/aggregation/mod.rs:277]: yep

File: src/aggregation/mod.rs (hunk 2 above)
Comment (PSeitz) [reply src/aggregation/mod.rs:285]: yep

Comment (PSeitz) [reply src/aggregation/mod.rs:285]: ah no this is an clippy error

File: src/aggregation/bucket/term_agg.rs [hunk 3]
```diff
@@ -423,41 +430,52 @@ impl SegmentTermCollector {
...
-                let key = std::str::from_utf8(key)
-                    .map_err(|utf8_err| DataCorruption::comment_only(utf8_err.to_string()))?;
-                if !dict.contains_key(key) {
-                    dict.insert(key.to_owned(), Default::default());
+                    let key = Key::Str(
+                        std::str::from_utf8(key)
+                            .map_err(|utf8_err| DataCorruption::comment_only(utf8_err.to_string()))?
+                            .to_string(),
+                    );
+                    dict.entry(key).or_insert_with(Default::default);
```
Comment (fulmicoton) [inline src/aggregation/bucket/term_agg.rs]: silly nitpick

```suggestion
                    dict.entry(key).or_default();
```

**Review by fulmicoton — APPROVED** [review]

**Pull request merged**